Improve invalid source path error messages

```
$ target/release/ufofmt 1 2 3 4 5
[ERROR] Read error in "4": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "5": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "3": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "1": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "2": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

With changes in #3:

```
target/release/ufofmt 1 2 3 4 5
[ERROR] "1" is not a valid UFO directory path
[ERROR] "2" is not a valid UFO directory path
[ERROR] "3" is not a valid UFO directory path
[ERROR] "4" is not a valid UFO directory path
[ERROR] "5" is not a valid UFO directory path
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -15,12 +15,33 @@ struct Opt {
 
     /// UFO source file paths
     #[structopt(help = "UFO source path(s)")]
-    ufopaths: Vec<String>,
+    ufopaths: Vec<PathBuf>,
 }
 
 fn main() {
     let argv = Opt::from_args();
 
+    // ~~~~~~~~~~~~~~~~~~~~~~
+    // UFO dir validity check
+    // ~~~~~~~~~~~~~~~~~~~~~~
+    let mut bad_ufo_path = false;
+    for ufo_testpath in &argv.ufopaths {
+        match ufo_testpath.exists() {
+            true => (),
+            false => {
+                let error_str = "[ERROR]".red().bold();
+                eprintln!("{} {:?} is not a valid UFO directory path", error_str, ufo_testpath);
+                bad_ufo_path = true;
+            }
+        }
+    }
+    if bad_ufo_path {
+        std::process::exit(1);
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Source formatting execution
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~
     let now = Instant::now();
     let results: Vec<u8> = argv.ufopaths.par_iter().map(|ufopath| format_ufo(ufopath)).collect();
     let duration = now.elapsed().as_millis();
@@ -38,21 +59,19 @@ fn main() {
 
 /// Read/write roundtrip through the norad library. Returns a 1 if an error was encountered
 /// and 0 if no error was encountered
-fn format_ufo(ufopathstr: &str) -> u8 {
-    let ufopath = PathBuf::from(ufopathstr);
-
-    match Font::load(&ufopath) {
-        Ok(ufo) => match ufo.save(&ufopath) {
+fn format_ufo(ufopath: &PathBuf) -> u8 {
+    match Font::load(ufopath) {
+        Ok(ufo) => match ufo.save(ufopath) {
             Ok(_) => 0,
             Err(e) => {
                 let error_str = "[ERROR]".red().bold();
-                eprintln!("{} Write error in {:?}: {}", error_str, &ufopath, e);
+                eprintln!("{} Write error in {:?}: {}", error_str, ufopath, e);
                 1
             }
         },
         Err(e) => {
             let error_str = "[ERROR]".red().bold();
-            eprintln!("{} Read error in {:?}: {}", error_str, &ufopath, e);
+            eprintln!("{} Read error in {:?}: {}", error_str, ufopath, e);
             1
         }
     }

